src/ui/Analytics/LiveAnalyticsClient.ts

```diff
@@ -151,11 +152,17 @@ export class LiveAnalyticsClient implements IAnalyticsClient {
     var customEvent = this.buildCustomEvent(actionCause, metaObject, element);
     this.triggerChangeAnalyticsCustomData('CustomEvent', metaObject, customEvent);
     this.checkToSendAnyPendingSearchAsYouType(actionCause);
-    Defer.defer(() => {
-      if (this.sendToCloud) {
-        this.endpoint.sendCustomEvent(customEvent);
-      }
-      $$(this.rootElement).trigger(AnalyticsEvents.customEvent, <IAnalyticsCustomEventArgs>{ customEvent: APIAnalyticsBuilder.convertCustomEventToAPI(customEvent) });
+    return new Promise((resolve) => {
```

Comment:
Do not create a `new Promise` when you can simply return another one created for you. It's an anti pattern otherwise.

This method can simply be : 

```
triggerEventChangeAlayticsData(...);
triggerEventConvertCustomEventToApi(...);
return this.endpoint.sendCustomEvent(...);
```